packages/scripts/scripts/build.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+const { sync: spawn } = require( 'cross-spawn' );
+const { sync: resolveBin } = require( 'resolve-bin' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	getCliArgs,
+	hasCliArg,
+	hasProjectFile,
+} = require( '../utils' );
+
+const hasWebpackConfig = hasCliArg( '--config' ) ||
+	hasProjectFile( 'webpack.config.js' ) ||
+	hasProjectFile( 'webpack.config.babel.js' );
+
+if ( hasWebpackConfig ) {
+	// Sets environment to production.
+	process.env.NODE_ENV = 'production';
```

Comment:
It's a bit strange to me for the script to be explicitly assigning an environment variable, though I think it's likely the most effective way to capture how most configurations would consider the current environment.

https://webpack.js.org/guides/production/#specify-the-mode

Reply:
It replicates what Gutenberg does as of today. There is a similar trick used for unit tests where `NODE_ENV` is forced to `test`.